packages/medusa/src/services/batch-job.ts

```diff
@@ -351,14 +368,35 @@ class BatchJobService extends TransactionBaseService<BatchJobService> {
         await this.update(batchJob, {
           result: {
             ...result,
-            errors: [...(result?.errors ?? []), reason],
+            errors: [...(result?.errors ?? []), error],
           },
         })
       }
 
       return await this.updateStatus(batchJob, BatchJobStatus.FAILED)
     })
   }
+
+  async prepareBatchJobForProcessing(
+    data: CreateBatchJobInput,
+    req: Request
+  ): Promise<CreateBatchJobInput> {
+    return await this.atomicPhase_(async () => {
+      let batchStrategy: AbstractBatchJobStrategy<never>
+      try {
+        const strategyResolver: StrategyResolver =
+          req.scope.resolve("strategyResolver")
+        batchStrategy = strategyResolver.resolveBatchJobByType(data.type)
+      } catch (e) {
+        throw new MedusaError(
+          MedusaError.Types.NOT_FOUND,
+          `Unable to find a BatchJob strategy with the type ${data.type}`
+        )
+      }
```

Comment:
fix: this is not necessary here as we are doing the check in the StrategyResolver